src/onnx/parse_resize.cpp

```diff
@@ -265,23 +265,38 @@ struct parse_resize : op_parser<parse_resize>
                                         ": dynamic input scale is not supported!");
 
                     arg_scale.visit([&](const auto& v) { vec_scale.assign(v.begin(), v.end()); });
-                    if(in_lens.size() != vec_scale.size())
-                    {
-                        MIGRAPHX_THROW("PARSE_" + opd.op_name +
-                                       ": ranks of input and scale are different!");
-                    }
-
-                    std::transform(in_lens.begin(),
-                                   in_lens.end(),
-                                   vec_scale.begin(),
-                                   out_lens.begin(),
-                                   [&](auto idx, auto scale) {
-                                       return static_cast<std::size_t>(idx * scale);
-                                   });
                 }
             }
         }
 
+        // scales still missing, must be an attribute
+        if(vec_scale.empty())
+        {
+            if(contains(info.attributes, "scales"))
```

Comment:
I think it would be better to first check if attribute exists or not. If it exist then, we can skip For loop at Line 219. 
"Resize" operator doesn't have `scales` attribute or input.  Therefore for the resize, it would always use `for` loop.